proxy.go

```diff
@@ -99,34 +104,37 @@ func serve(servConn io.ReadWriteCloser, proto, addr string, results chan error)
 			var conn, stream net.Conn
 			conn, err = l.Accept()
 			if err != nil {
-				return
+				break
 			}
 
 			stream, err = session.Open()
 			if err != nil {
-				return
+				break
 			}
 
-			go proxyConn(conn, stream)
+			go proxyConn(conn, stream, wg)
```

Comment:
This is potentially racy. Usually, we call `wg.Add()` before to call into the go routine calling `wg.Done()`.
The current go routine here might reach `wg.Done()` before the subsequent go routines will call into `wg.Add()`.

Reply:
LGTM now !